kcidb/argparse.py

```diff
@@ -0,0 +1,298 @@
+"""KCIDB argument parsing"""
+
+import math
+import re
+import os
+import argparse
+import logging
+import dateutil.parser
+try:  # Python 3.9
+    from importlib import metadata
+except ImportError:  # Python 3.6
+    import importlib_metadata as metadata
+import kcidb.io as io
+from kcidb.misc import logging_setup
+
+
+# A dictionary of names of logging levels and their values
+LOGGING_LEVEL_MAP = {
+    name: value
+    for name, value in logging.__dict__.items()
+    if name.isalpha() and name.isupper() and isinstance(value, int) and value
+}
+# Logging level disabling all logging
+LOGGING_LEVEL_MAP["NONE"] = max(LOGGING_LEVEL_MAP.values()) + 1
+# Sort levels highest->lowest
+# I don't see it, pylint: disable=unnecessary-comprehension
+LOGGING_LEVEL_MAP = {
+    k: v
+    for k, v in sorted(LOGGING_LEVEL_MAP.items(),
+                       key=lambda i: i[1], reverse=True)
+}
```

Comment:
These `LOGGING_*` symbols shouldn't be here, and they definitely shouldn't me simply copied here. They should be defined in one place only, to avoid desynchronization. I see that both the `kcidb.misc` and `kcidb.argparse` modules use those. To avoid both this copying and a circular import, we need to move all symbols starting with `LOGGING_` or `logging_` to their own module called `kcidb.logging` (in `kcidb/logging.py` file).

After moving the symbols should naturally lose their `LOGGING_`/`logging_` prefixes, so e.g. `kcidb.misc.LOGGING_LEVEL_MAP` becomes `kcidb.logging.LEVEL_MAP`, and `kcidb.misc.logging_setup` becomes `kcidb.logging.setup`, and so on.

This change would better be done **before** extracting the argparse modules, in a separate commit.